src/Simulation/Core/ArgumentValue.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.Quantum.Simulation.Core;
+using Core = Microsoft.Quantum.Simulation.Core;
+
+namespace Microsoft.Quantum.Runtime
+{
+    /// <summary>
+    /// The value of an argument to a QIR callable is a discriminated union of the argument types.
+    /// </summary>
+    public abstract class ArgumentValue
```

Comment:
Are the types listed in this class (and in ArgumentType) related to/dictated by what the bond schema for the entry point arguments looks like? If so, then they will change as part of eliminating the dependency on the Q# compiler data structures and should hence not be public.

Reply:
They are related to the Bond schema but the schema is dictated by what type of arguments are supported as entry point.

Similarly to this [comment](https://github.com/microsoft/qsharp-runtime/pull/761#discussion_r665195518) we should have a conversation about this being public before merging.

Reply:
Resolved per the other thread.